toolchest_client/files/tests/test_general.py

```diff
@@ -0,0 +1,50 @@
+import filecmp
+import os
+import pathlib
+
+import pytest
+
+from .. import assert_exists, check_file_size, files_in_path, sanity_check
+
+THIS_FILE_PATH = os.path.normpath(pathlib.Path(__file__).parent.resolve())
+
+
+def test_small_file():
+    small_file_path = f"{THIS_FILE_PATH}/data/very_small_file.txt"
+    with pytest.raises(ValueError):
+        sanity_check(small_file_path)
+
+def test_files_in_path():
+    tmp_dir = f"{THIS_FILE_PATH}/tmp"
+    tmp1 = f"{tmp_dir}/tmp1"
+    tmp2 = f"{tmp_dir}/tmp2"
+    sub_dir = f"{tmp_dir}/sub_dir"
+    tmp3 = f"{sub_dir}/tmp3"
+    file_paths = [tmp1, tmp2, tmp3]
+    for dir in [tmp_dir, sub_dir]:
+        if not os.path.exists(dir):
+            os.mkdir(dir)
+    for file in file_paths:
+        open(file, "w")
+    file_paths = sorted([os.path.normpath(x) for x in file_paths])
+
+    assert sorted([os.path.normpath(x) for x in files_in_path(tmp_dir)]) == file_paths
+
+    for file in file_paths:
+        os.remove(file)
+    os.rmdir(sub_dir)
+    os.rmdir(tmp_dir)
```

Comment:
Not sure if there's a better way to do this; if the `assert` fails, then the `tmp` files will remain in the `files/tests` directory once tests end.

Reply:
You could add:
a) a pytest fixture that runs before and after each test
b) a try-catch-finally

(a) would be pretty non-generalizable, (b) seems more apt